frontend-web/webclient/app/Applications/Jobs/JobsBrowse.tsx

```diff
@@ -308,4 +352,89 @@ const JOB_STATE_AND_ICON_COLOR_MAP: Record<JobState, [IconName, ThemeColor]> = {
     CANCELING: ["heroXMark", "errorMain"]
 };
 
+export function ProjectMemberFilter({onSelect}: {onSelect: (username: string) => void}) {
+    const [activeUser, setActiveUser] = React.useState("");
+    const [filter, setFilter] = React.useState("");
+
+    const p = useProject();
+    const project = p.fetch();
+    const projectId = useProjectId();
+    const members = project.status.members?.map(it => it.username) ?? [Client.username!];
+    const avatars = useAvatars();
+
+    const setFilterMember = React.useCallback((member: string) => {
+        setFilter("");
+        onSelect(member);
+        setActiveUser(member);
+    }, []);
+
+    React.useEffect(() => {
+        setFilterMember("")
+    }, [projectId]);
+
+    return <ClickableDropdown
+        arrowkeyNavigationKey="data-user"
+        onSelect={el => {
+            const user = el?.getAttribute("data-user");
+            if (user) setFilterMember(user);
+        }}
+        paddingControlledByContent
+        colorOnHover={false}
+        useMousePositioning
+        trigger={
+            <Flex my="auto">
+                <Icon size="12px" my="auto" mr="10px" name="user" color="textPrimary" />
+                <Text>Created by </Text> {activeUser ? <Flex pr="8px">
+                    <UserAvatar width={"24px"} height={"24px"} avatar={avatars.avatar(activeUser)} /> <Truncate title={activeUser} fontSize={"16px"}>{activeUser}</Truncate>
+                </Flex>
+                    : null}
+            </Flex>
+        }
+    >
+        <Flex>
+            <Input
+                autoFocus
+                className={FilterInputClass}
+                placeholder="Search for a project..."
+                defaultValue={filter}
+                onClick={stopPropagationAndPreventDefault}
+                enterKeyHint="enter"
+                onKeyDown={e => {
+                    // Note(Jonas): Not reached for some reason?
+                    if (["Escape"].includes(e.code) && e.target["value"]) {
+                        setFilter("");
+                        e.target["value"] = "";
+                    }
+                    // Note(Jonas): Stop keypropagation like `Delete`.
+                    e.stopPropagation();
+                }}
+                onKeyUp={e => {
+                    e.stopPropagation();
+                    setFilter("value" in e.target ? e.target.value as string : "");
+                }}
+                type="text"
+            />
+            <Relative right="24px" top="5px" width="0px" height="0px">
+                <Icon name="search" />
+            </Relative>
+        </Flex>
+        <Box maxHeight={"450px"} overflowY="scroll">
+            {activeUser ? <Flex className={HoverClass} height={"32px"} onClick={() => setFilterMember("")}><Icon ml="12px" mt="8px" size="16px" color="errorMain" name="close" /> <Truncate ml="11px" style={{alignContent: "center"}} fontSize={"16px"}>Clear</Truncate></Flex> : null}
```

Comment:
A bit too long for my taste. I think something like this would be better.

```suggestion
            {activeUser ?
                <Flex
                    className={HoverClass}
                    height={"32px"}
                    onClick={() => setFilterMember("")}
                >
                    <Icon ml="12px" mt="8px" size="16px" color="errorMain" name="close" />
                    <Truncate ml="11px" style={{alignContent: "center"}} fontSize={"16px"}>
                        Clear
                    </Truncate>
                </Flex> :
                null
            }
```